retry/retry.go

```diff
@@ -82,7 +82,9 @@ func (r *Retrier) Retry(f func() (interface{}, error)) (interface{}, error) {
 		}
 
 		time.Sleep(next)
-		r.notifyRetry(err, numTries)
+		if len(r.notifyRetryFuncs) > 0 {
```

Comment:
I don't think you need this check - since if notifyRetryFuncs is a blank list, it just won't call anything.